src/main/java/edu/ie3/datamodel/io/csv/HierarchicFileNamingStrategy.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.io.csv;
+
+import edu.ie3.datamodel.models.UniqueEntity;
+import edu.ie3.datamodel.models.timeseries.TimeSeries;
+import edu.ie3.datamodel.models.timeseries.TimeSeriesEntry;
+import edu.ie3.datamodel.models.value.Value;
+import java.util.Optional;
+import org.apache.commons.io.FilenameUtils;
+
+/** A file naming strategy, that takes hierarchic order of sub folders into account. */
+public class HierarchicFileNamingStrategy extends FileNamingStrategy {
+  private final FileHierarchy hierarchy;
+
+  public HierarchicFileNamingStrategy(String prefix, String suffix, FileHierarchy hierarchy) {
+    super(prefix, suffix);
+    this.hierarchy = hierarchy;
+  }
+
+  public HierarchicFileNamingStrategy(FileHierarchy hierarchy) {
+    this.hierarchy = hierarchy;
+  }
+
+  public HierarchicFileNamingStrategy(String prefix, FileHierarchy hierarchy) {
+    super(prefix);
+    this.hierarchy = hierarchy;
+  }
+
+  /**
+   * Determines the filename for a certain class by composing the hierarchic sub directory structure
+   * provided by {@link this#hierarchy} with the actual file name given by {@link super}. If one of
+   * both components cannot be determined, an empty {@link Optional} is returned.
+   *
+   * @param cls The class to define the file name for.
+   * @return The file name including the hierarchic sub directory structure
+   */
+  @Override
+  public Optional<String> getFileName(Class<? extends UniqueEntity> cls) {
+    /* Get the file name */
+    Optional<String> maybeFilename = super.getFileName(cls);
+    String fileName;
+    if (!maybeFilename.isPresent()) {
+      logger.debug("Cannot determine file name for class '{}'.", cls);
+      return Optional.empty();
+    } else {
+      /* Make sure, that the file name does not start with a file separator */
+      fileName = maybeFilename.get().replaceFirst("^[\\\\/]", "");
```

Comment:
Pls consider addressing this sonarqube code smell by defining a private constant for the replacement regex.